numpy/core/src/umath/ufunc_object.c

```diff
@@ -2336,10 +2408,11 @@ PyUFunc_GeneralizedFunction(PyUFuncObject *ufunc,
         /* Except for when it belongs to this output */
         if (i >= nin) {
             int dim_offset = ufunc->core_offsets[i];
-            int num_dims = ufunc->core_num_dims[i];
+            int num_dims = core_num_dims[i];
+
             /* Fill in 'iter_shape' and 'op_axes' for this output */
             for (idim = 0; idim < num_dims; ++idim) {
-                iter_shape[j] = core_dim_sizes[
+                iter_shape[j] = keepdims ? 1 : core_dim_sizes[
```

Comment:
Possibly clearer with an `if`, because then that gives you somewhere to add a comment explaining _why_ it is 1,

Reply:
I expanded the comment above the loop (which is not very useful as is...)